### PR TITLE
Tomoz/superspreading

### PR DESCRIPTION
**Pull request by tozanski**

### CODE REVIEW

File: julia/src/Simulation/params/spreading.jl [hunk 1]
```diff
@@ -0,0 +1,27 @@
+using Distributions
+using Random
+
+struct SpreadingParams
```
Comment (eMaerthin) [inline julia/src/Simulation/params/spreading.jl:4]: maybe call it `TruncatedParetoParams`?

File: julia/src/Simulation/params/spreading.jl [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  dict[prefix*"dist"] = params(p.dist)
+  dict[prefix*"values"] = p.spreading
+  nothing
+end
+
+function show(io::IO, p::SpreadingParams)
+  print(io, "Spreading params for ", length(p), " individuals, distribution=", p.dist)
+end
+
+spreading(p::SpreadingParams, source_id::Integer) = p.spreading[source_id]
```
Comment (eMaerthin) [inline julia/src/Simulation/params/spreading.jl:26]: maybe call it `truncatedparetospreading`?

File: julia/src/Simulation/infection_kernels.jl [hunk 3]
```diff
@@ -131,6 +133,8 @@ function enqueue_transmissions!(state::SimState, ::Type{Val{FriendshipContact}},
 
   total_infection_rate = (end_time - start_time) * params.friendship_kernel_params.kernel_constant * socialcompetence(params, source_id)
 
+  total_infection_rate *= spreading(params, source_id)
```
Comment (eMaerthin) [inline julia/src/Simulation/infection_kernels.jl:136]: maybe each kernel should have its own `spreading` or at least flag to turn it on or off?

File: julia/src/Simulation/simparams.jl [hunk 4]
```diff
@@ -127,7 +131,11 @@ function make_params(
   testing_time::Float64=1.0,
 
   phone_tracking_usage::Real=0.0,
-  phone_detection_delay::Real=0.25
+  phone_detection_delay::Real=0.25,
+
+  spreading_alpha::Union{Nothing,Real}=4,
```
Comment (eMaerthin) [inline julia/src/Simulation/simparams.jl:136]: I suggest to use a default value that changes nothing compared to previous vanilla simulation, so `spreading_alpha::Union{Nothing,Real}=nothing,`

File: julia/src/Simulation/simparams.jl [hunk 5]
```diff
@@ -214,5 +230,6 @@ function saveparams(dict, p::SimParams)
   saveparams(dict, p.progressions, "progressions/")
   nothing==p.hospital_kernel_params || saveparams(dict, p.hospital_kernel_params, "hospital/")
   nothing==p.phone_tracking_params || saveparams(dict, p.phone_tracking_params, "phone_tracking/")
+  nothing==p.spreading_params || saveparams(dict, p.spreading_params, "super_spreading")
```
Comment (eMaerthin) [inline julia/src/Simulation/simparams.jl:233]: maybe rename `super_spreading` to `spreading`? or rename `spreading_params` to `super_spreading_params`?

File: julia/src/intermediate_cli.jl [hunk 6]
```diff
@@ -53,6 +53,11 @@ function read_params(json, rng::AbstractRNG)
     modulation["function"], NamedTuple{Tuple(Symbol.(keys(params)))}(values(params))
   end
 
+  spreading = get(json, "super_spreading", nothing)
+  spreading_alpha = isnothing(spreading) ? nothing : spreading["alpha"]
```
Comment (eMaerthin) [inline julia/src/intermediate_cli.jl:57]: maybe 
```suggestion
  spreading_alpha = isnothing(spreading) ? nothing : get(spreading, "alpha", 4)
```
?

File: example/example.json [hunk 7]
```diff
@@ -24,6 +24,11 @@
       "limit_value": 1.0
     }
   },
+  "super_spreading" : {
```
Comment (eMaerthin) [inline example/example.json:27]: I prefer to use `spreading` or rename the struct in other part of code (try to use consistent names)

File: julia/src/Simulation/params/spreading.jl (hunk 1 above)
Comment (tozanski) [reply julia/src/Simulation/params/spreading.jl:4]: They are parameters for Superspreading not for Truncated Pareto Distribution 🤷

File: julia/src/Simulation/params/spreading.jl (hunk 2 above)
Comment (tozanski) [reply julia/src/Simulation/params/spreading.jl:26]: in julia the function names are encouraged to be simple - the name clashes are not as dangerous as in languages without multiple dispatch.

File: julia/src/Simulation/simparams.jl (hunk 4 above)
Comment (tozanski) [reply julia/src/Simulation/simparams.jl:136]: good catch, that was the intention

File: julia/src/intermediate_cli.jl (hunk 6 above)
Comment (tozanski) [reply julia/src/intermediate_cli.jl:57]: I would prevent the defalut value for this parameter fro being defined as it is not clear what it should be.

File: julia/src/Simulation/simparams.jl (hunk 5 above)
Comment (tozanski) [reply julia/src/Simulation/simparams.jl:233]: fair enough, let's fix on spreading then

**Pull request merged**